packages/core/src/new-api/internal/reconciliation/execution-state-resolver.ts

```diff
@@ -14,12 +15,59 @@ import {
   StaticCallExecutionState,
 } from "../types/execution-state";
 import { isAddress } from "../utils";
+import { replaceWithinArg } from "../utils/replace-within-arg";
 
 import { ReconciliationContext } from "./types";
 import { resolveModuleParameter, safeToString } from "./utils";
 
 // TODO: consider merging this into the execution state map
 export class ExecutionStateResolver {
+  public static resolveArgsFromExectuionState(
+    constructorArgs: ArgumentType[],
+    context: ReconciliationContext
+  ): ArgumentType[] {
+    return replaceWithinArg<ArgumentType>(constructorArgs, {
+      bigint: (bi) => bi.toString(),
+      future: (f) => {
+        if (!isContractFuture(f)) {
```

Comment:
There's also static call and event argument futures

Reply:
I like to pretend they don't exist.

Reply:
I have added tests and conditionals for these two cases. It is the last commit. 

Note this required two refactorings worth paying attention to:

1. Add a SolidityParameterType (identical to ModuleParameterType)
2. Add a result to read event state with type SolidityParameterType

It seemed weird to use ModuleParameterType so far outside a module, though the answer may be to rename it. I also needed a result for ReadEventArgumentState, I am happy to make it a more complex object depending on how we expand out the event api.